lib/ultrajsonenc.c

```diff
@@ -261,11 +279,19 @@ static int Buffer_EscapeStringUnvalidated (JSONObjectEncoder *enc, const char *i
 
 static int Buffer_EscapeStringValidated (JSOBJ obj, JSONObjectEncoder *enc, const char *io, const char *end)
 {
+  Buffer_Realloc(enc, RESERVE_STRING(end - io));
```

Comment:
Is this supposed to be realloc, not reserve? If so, what's the reasoning behind that?